File: CsharpProjects/myLeetCode/Linked_List/206. Reverse Linked List.cs
// 206. Reverse Linked List
// https://leetcode.com/problems/reverse-linked-list/

using System;

public class Solution {
    public ListNode ReverseList(ListNode head) {
        ListNode prev = null;
        ListNode current = head;

        while (current != null) {
            ListNode nextTemp = current.next;
            current.next = prev;
            prev = current;
            current = nextTemp;
        }
        
        return prev;
    }
}

public class Program {
    public static void Main() {
        int[] nums = {1, 2, 3, 4, 5};
        ListNode head = ListHelpers.ConvertArrayToList(nums);

        Solution solution = new Solution();
        ListNode result = solution.ReverseList(head);

        Console.Write("Output: ");
        while (result != null) {
            Console.Write(result.val + " ");
            result = result.next;
        }
    }
}

File: C++/代码随想录/Linked_List/206. Reverse Linked List.py
from typing import Optional

class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        prev = None
        current = head
        while current:
            next_temp = current.next
            current.next = prev
            prev = current
            current = next_temp
        return prev

def printList(head):
    values = []
    while head:
        values.append(head.val)
        head = head.next
    return values

sol = Solution()

# 创建一个链表 [1, 2, 3, 4, 5]
head = ListNode(1, ListNode(2, ListNode(3, ListNode(4, ListNode(5)))))

# 反转链表
print(printList(sol.reverseList(head)))  # 应该输出 [5, 4, 3, 2, 1]
